tests/test_job.py

```diff
@@ -212,7 +212,8 @@ async def coro2():
     await scheduler.spawn(coro2())
 
     await fut
-    await job.wait()
```

Comment:
If you look at a similar test above, on line 101, you'll see the code expects a return value there, but does not here.

The only reason we don't get a return value here is the brief yield to the event loop from `await fut`. I don't see anyway a user should be able to reliably predict whether this method will return `None` or the result of the task.

The change above to return the task value if it is already completed seems a lot more reasonable and more consistent with general asyncio usage (e.g. `await task`).